### ISSUE TITLE
WCAG issues from the plugins evaluation

### ISSUE THREAD

**Issue by jolevesq** [fgpv-vpgf/fgpv-vpgf#3892]
We did a WCAG evaluation for the plugins and this is what we get. An issue has been raise for thing specific to plugins (https://github.com/fgpv-vpgf/contributed-plugins/issues/147). Below are the thing more specific to the viewer. Some stuff may not be doable but I have listed them below to keep the info.

1. Export button in the menu is missing aria label
2. The side panel of data is visually structured but is not semantically structured
    * Add a heading 4 and definition lists for the JSON type relationships. CSS can be used to present the dt and dd on the same line
    *![image.png](https://images.zenhubusercontent.com/5695178c638e718c99600115/bd767283-377f-4f63-ba17-36d032519293)
3. Interactive controls don't have a border or any identifier that passes the minimum contrast requirements of 3:1 with the background
    * Make the buttons and icons darker. Make it at least #929292
4.  Focus ring (dotted/solid line around the control that has focus) has insufficient contrast around certain interactive controls as user tabs through the page. | Interactive elements do not show visual focus when it receives tab focus. | A keyboard user tabbing through the page will not know when focus is on controls that don't have a focus ring.
    * ![image.png](https://images.zenhubusercontent.com/5695178c638e718c99600115/9a7c5ec9-0ef8-4f54-a28b-52cb58877549)
    versus
    * ![image.png](https://images.zenhubusercontent.com/5695178c638e718c99600115/989daec3-a9af-4260-ad72-2c32f53936c7)
    * The anchor outline:none appears to over ride the :focus class to remove the visual focus when the tab key is used to access the element.. To solve this, remove outline:none from the anchor classAnd focus:noneNote: if these are user agent Style sheet settings, simply override them with visible outline values in the CSS. In the example solution to the left its a grey outline. It could be and dark color such as red, dark blue, green etc... |   | Delete this {outline: none} This code will ensure the focus is always visible and has sufficient contrast regardless of the background.*:focus { box-shadow: inset 0 0 0 1px black, 0 0 3px 1px white !important;outline: dashed !important;}Live example linked to the right
5.  The arrow keys have a very coarse adjustment and it is difficult to set focus on the small dots which don't increase in size when the user zooms in.
    *  Provide a modulating key which makes the cursor move much more granular. I suggest trapping the CONTROL key and when the user holds it down and presses the arrows the cursor moves much more granularly, (slower).There could also be a modulating key to make it go faster. For instance, when the ALT key is held sown, pressing the arrow keys is much less granular and the cursor moves quicker.
6. Hamburger menu stays open if an item is activated with mouse but closes if a keyboard is used.  With mouse sometimes it stays open sometimes it stays closed.
    *  Leave the side panel open until the user dismisses it with "close" or perhaps an OK button.  If a button on the side panel opens a dialog, then the normal behavior is to return the user to the button that launched the dialog when the dialog is closed. An example would be the export button which opens a dialog box and closes the side menu, but doesn't return the user back to the export button when the dialog is closed. A more natural behavior would be to leave the side panel open when launching the dialog (it can be greyed out with the rest of the background.) Then return the user to the export button when the dialog is closed.
7. The mechanism to go into keyboard mode is visually presented as a dialog box, with a greyed out background. But is doesn't act like one and doesn't present the proper roles and states
    * ![image.png](https://images.zenhubusercontent.com/5695178c638e718c99600115/ea19d1e8-6541-4643-96bc-35e167152b04)
    * There are a number of fixes1) Don't have it automatically popup. That is confusing.2) Provide an invisible button that shows up on focus that says "Open keyboard Instructions"3) This would open a a true dialog, where the focus is placed inside it4) Tab focus would remain inside it, and it is dismissed either by hitting then OK button, the X button or pressing escape ESC key.5) a model for the modal is to the right. It provides the HTML, CSS, JavaScript and ARIA. The instructions could then be more comprehensive to describe how to draw on the map etc. if that is appropriate for the functionality of the map. 
    * https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html
8. Modal dialog is missing its roles states and name 
    * When the side panel is open the background is greyed out visually which makes it a modal. It should announce as such and should keep keyboard focus inside it. Add to the dialog container  role="dialog" aria-modal="true" aria-labelledby="[id of heading for dialog]" 
    * https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html
9.  There is a long/lat coordinates provided on the screen to provide the location of the mouse, but this is not provided to the keyboard user.
10. The legend and map don't have an alternative way to present the data, except in color

**Comment by dan-bowerman** [comment]
Putting a large estimate on this one for the moment - it will need to be broken down into smaller tasks soon.

**Comment by dan-bowerman** [comment]
Closing this, the issues noted above reflect the breakdown of this request into individual work items.
